lib/src/arithmetic/floating_point/floating_point_sqrt_simple.dart

```diff
@@ -0,0 +1,102 @@
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Indentifier: BSD-3-Clause
+//
+// floating_point_sqrt.dart
+// An abstract base class defining the API for floating-point square root.
+//
+// 2025 March 4
+// Authors: James Farwell <james.c.farwell@intel.com>,
+//Stephen Weeks <stephen.weeks@intel.com>,
+//Curtis Anderson <curtis.anders@intel.com>
+
+import 'package:rohd/rohd.dart';
+import 'package:rohd_hcl/rohd_hcl.dart';
+
+/// An square root module for FloatingPoint values
+class FloatingPointSqrtSimple<FpType extends FloatingPoint>
+    extends FloatingPointSqrt<FpType> {
+  /// Square root one floating point number [a], returning results
+  /// [sqrtR] and [error]
+  FloatingPointSqrtSimple(super.a,
+      {super.clk,
+      super.reset,
+      super.enable,
+      super.name = 'floatingpoint_square_root_simple'})
+      : super(
+            definitionName: 'FloatingPointSquareRootSimple_'
+                'E${a.exponent.width}M${a.mantissa.width}') {
+    final outputSqrt = FloatingPoint(
```

Comment:
Noticed the 'normal' tests:  are subnormals supported?  If not, you should check and throw.

Reply:
Good catch! I forgot that! I'll fix that this weekend though I may need pointers on how to do that

Reply:
I think you have an error flag, so you would just need to assign it to ~fp.isNormal();

Reply:
I decided to add another test for the error flag on the output, and to gate the fixed point using a DeNorm flag. This is passing tests and is pushed